src/constructs/ses/index.ts

```diff
@@ -0,0 +1,47 @@
+import { Duration } from "aws-cdk-lib";
+import type { EmailIdentityProps } from "aws-cdk-lib/aws-ses";
+import { EmailIdentity, Identity } from "aws-cdk-lib/aws-ses";
+import { GuAppAwareConstruct } from "../../utils/mixin/app-aware-construct";
+import type { AppIdentity, GuStack } from "../core";
+import { GuCname } from "../dns";
+
+export interface GuEmailIdentityProps extends Omit<EmailIdentityProps, "identity">, AppIdentity {
+  domainName: string;
+}
+
+/**
+ * A construct to create an SES email identity.
+ * It also creates the required DNS records to verify the identity.
+ * @see https://docs.aws.amazon.com/ses/latest/DeveloperGuide/verify-domains.html
+ * @see https://docs.aws.amazon.com/ses/latest/DeveloperGuide/send-email-identity.html
+ *
+ * @example
+ * new GuEmailIdentity(stack, "MyEmailIdentity", {
+ *  domainName: "my-service.gutools.co.uk",
+ *  app: "test",
+ * });
+ **/
+export class GuEmailIdentity extends GuAppAwareConstruct(EmailIdentity) {
+  constructor(scope: GuStack, id: string, props: GuEmailIdentityProps) {
+    const { app, domainName } = props;
+    if (!domainName.endsWith("gutools.co.uk")) {
+      throw new Error(
+        `Auto verification is only supported for gutools.co.uk domains. ${domainName} is not a gutools.co.uk domain.`
+      );
```

Comment:
Why are we limited to this domain?

Reply:
because we're using GuCname under the hood

Reply:
Ah. We've recently broadened the domains available with this tooling. See https://github.com/guardian/dns-validation-lambda/pull/43/files for a complete list[^1]. It's not massively publicised yet because we still need to properly document use-cases for the domains.

Maybe it's worth adding an inline todo comment along the lines of:

> TODO: expand this as https://github.com/guardian/cfn-private-resource-types/tree/main/dns/guardian-dns-record-set-type does

?

[^1]: The link is to an unrelated repository, however the credentials used between that and the DNS lambdas are the same.